src/controller/buffer-controller.ts

```diff
@@ -218,13 +270,17 @@ class BufferController extends EventHandler {
     let pending = this.segments.reduce((counter, segment) => (segment.parent === parent) ? counter + 1 : counter, 0);
 
     // this.sourceBuffer is better to use than media.buffered as it is closer to the PTS data from the fragments
-    let timeRanges = {};
-    const sourceBuffer = this.sourceBuffer;
-    for (let streamType in sourceBuffer) {
-      timeRanges[streamType] = sourceBuffer[streamType].buffered;
+    let timeRanges: any = {};
```

Comment:
can use `TimeRanges` here
```suggestion
    const timeRanges: { [streamType: keyof SourceBufferSet]: TimeRanges };
```